Content.Server/Interaction/InteractionSystem.cs

```diff
@@ -432,6 +475,44 @@ public async Task InteractUsing(IEntity user, IEntity used, IEntity target, Enti
             await InteractDoAfter(user, used, target, clickLocation, true);
         }
 
+        /// <summary>
+        ///     Alternative interactions on an entity. 
+        /// </summary>
+        /// <remarks>
+        ///     Uses the context menu verb list, and acts out the first verb marked as an alternative interaction. Note
+        ///     that this does not have any checks to see whether this interaction is valid, as these are all done in <see
+        ///     cref="UserInteraction(IEntity, EntityCoordinates, EntityUid, bool)"/>
+        /// </remarks>
+        public void AltInteract(IEntity user, IEntity target)
+        {
+            // TODO VERB SYSTEM when ECS-ing verbs and re-writing VerbUtility.GetVerbs, maybe sort verbs by some
+            // priority property, such that which verbs appear first is more predictable?.
+
+            // Iterate through list of verbs that apply to target. We do not include global verbs here. If in the future
+            // alt click should also support global verbs, this needs to be changed.
+            foreach (var (component, verb) in VerbUtility.GetVerbs(target))
+            {
+                // Check that the verb marked as an alternative interaction?
+                if (!verb.AlternativeInteraction)
+                    continue;
+
+                // Can the verb be acted out?
+                if (!VerbUtility.VerbAccessChecks(user, target, verb))
+                    continue;
+
+                // Is the verb currently enabled?
+                var verbData = verb.GetData(user, component);
+                if (verbData.IsInvisible || verbData.IsDisabled)
+                    continue;
```

Comment:
surely this can be pulled out into some function for checking if a verb is able to be done or not? (perhaps in VerbUtility?) i imagine the actual verb activating code also uses this

Reply:
The way it's done currently is a bit odd. Verbs are defined in components, and each one defines a GetData() function, which can be called to determine whether the verb should be visible or not. E.g. ChemMaster marks eject as invisible if no beaker is present.

`VerbUility.GetVerbs()` just gets a list of all verbs that exist in an entity's components, without checking whether the verbs are even 'visible', but also provides global checks for verbs via `VerbAccessChecks()`, which checks if the target is in range and unobstructed, but doesn't run them when collecting verbs. VerbSystem, then actually calls `VerbAccessChecks()` and the `GetData()` functions to checks whether the verb should even be shown.

I think all the verb check functions should just be performed by the VerbUtility.GetVerbs() function (or maybe GetValidVerbs()? But I'm not gonna bother do that here because this has to be re-written anyway then ECSed. I guess that whenever verbs are ECS-ed you would instead have some sort of a AssembleVerbListEvent that various systems subscribe to. They would then conditionally add their verbs if and only if they are actually executable?